tests/functional/builtins/codegen/test_slice.py

```diff
@@ -536,3 +536,34 @@ def test_slice_buffer_oob_reverts(bad_code, get_contract, tx_failed):
     c = get_contract(bad_code)
     with tx_failed():
         c.do_slice()
+
+
+def test_slice_adhoc_locs_evals_once_side_effects(get_contract):
+    code = """
+@external
+def ret10_slice() -> Bytes[10]:
+    b: Bytes[32] = concat(convert(65, bytes32), b'')
+    c: Bytes[10] = slice(b, 31, 1)
+    return c
+    """
+
+    c = get_contract(code)
+    assert c.ret10_slice() == b"A"
+
+
+def test_sqrt_evals_once_side_effects(get_contract):
```

Comment:
no need for the name to have both "eval once" and "side effects"

Reply:
renamed